orte/mca/odls/singularity/odls_singularity_module.c

```diff
@@ -0,0 +1,828 @@
+        cd->argv[1] = NULL;
+    }
+
+    /* Set signal handlers back to the singularity.  Do this close to
+       the exev() because the event library may (and likely will)
+       reset them.  If we don't do this, the event library may
+       have left some set that, at least on some OS's, don't get
+       reset via fork() or exec().  Hence, the launched process
+       could be unkillable (for example). */
+
+    set_handler_singularity(SIGTERM);
+    set_handler_singularity(SIGINT);
+    set_handler_singularity(SIGHUP);
+    set_handler_singularity(SIGPIPE);
+    set_handler_singularity(SIGCHLD);
+
+    /* Unblock all signals, for many of the same reasons that we
+       set the singularity handlers, above.  This is noticable on
+       Linux where the event library blocks SIGTERM, but we don't
+       want that blocked by the launched process. */
+    sigprocmask(0, 0, &sigs);
+    sigprocmask(SIG_UNBLOCK, &sigs, 0);
+
+    /* take us to the correct wdir */
+    if (NULL != cd->wdir)
+    {
+        if (0 != chdir(cd->wdir))
+        {
+            send_error_show_help(write_fd, 1,
+                                 "help-orterun.txt",
+                                 "orterun:wdir-not-found",
+                                 "orted",
+                                 cd->wdir,
+                                 orte_process_info.nodename,
+                                 (NULL == cd->child) ? 0 : cd->child->app_rank);
+            /* Does not return */
+        }
+    }
+
+    /* Exec the new executable, i.e., fire up the Singularity container */
+
+    // We need to parse the environment of the application because there is important
+    // information in it to be able to set everything up
+    i = 0;
+    while (cd->env && cd->env[i])
+    {
+        if (strncmp(cd->env[i], singularity_exec_argc_env_var_name, strlen(singularity_exec_argc_env_var_name)) == 0)
+        {
+            exec_args = &cd->env[i][strlen(singularity_exec_argc_env_var_name) + 1]; // We do not want the "="
+            break;
+        }
+
+        i++;
+    }
+
+    singularity_bin_path = opal_path_findv("singularity", X_OK, cd->env, NULL);
+
+    // We need to prepend cd->cmd to cd->argv because in our case, cd->cmd is the
+    // path to the container's image and the implicit command is "singularity"
+
+    opal_argv_append_nosize(&cmd_args, singularity_bin_path);
+    opal_argv_append_nosize(&cmd_args, "exec");
+    if (exec_args != NULL)
+    {
+        char **args = opal_argv_split(exec_args, ' ');
+        i = 0;
+        while (args[i])
+        {
+            opal_argv_append_nosize(&cmd_args, args[i]);
+            i++;
+        }
+    }
+    i = 0;
+    while (cd->argv[i])
+    {
+        opal_argv_append_nosize(&cmd_args, cd->argv[i]);
+        i++;
+    }
+
+    ///home/gvallee/.sympi/install_singularity-master/bin/
+    execve("/home/gvallee/.sympi/install_singularity-master/bin/singularity", cmd_args, cd->env);
```

Comment:
Was this really meant to be committed ?

Reply:
Ah, good catch - I will bring over the Singularity configury from an OMPI version when we supported it

Reply:
That is embarrassing, sorry about that.
I will fix this through another PR. Note that the goal here is to scan the PATH of the app to find the version of Singularity that is specified through it. Requiring to re-configure PRRTE every time we want to change the version of Singularity is something I want to avoid. I will also add comments.